Sources/MongoSwift/MongoClient.swift

```diff
@@ -168,8 +170,23 @@ public class MongoClient {
     /// Default maximum size for connection pools created by this client.
     internal static let defaultMaxConnectionPoolSize = 100
 
-    /// Indicates whether this client has been closed.
-    internal private(set) var isClosed = false
+    /// Indicates the state of this `MongoClient`.
+    internal enum State {
```

Comment:
I don't really have strong feelings on this one. I think you summed the situation up well, and its just to decide whether we want consistency across drivers or across ecosystem members. 

As a related topic: do we want to consider the naming of `MongoCursor`'s `isAlive`/`kill` names in these terms? From brief examination, it looks like most drivers use `isOpen`/`isClosed`/`close` instead (Java, Go, Ruby were the ones I looked at).

Reply:
for the sake of moving forward on this, it seems like @mbroadst feels the most strongly so I say we go back to open/closed naming for the client.

regarding cursors, I believe I copied the `alive` thing from pymongo`. they use `close` for the closing method.  mongoc uses `mongoc_cursor_more` and `mongoc_cursor_destroy`. 
node also uses `close`, and looks like they have an `end` event that gets emitted so I think that's how you can tell if there's potential for more data? possibly we can defer a decision on this to next week when I'd like to work out the potential API changes to line up with Rust (I opened [SWIFT-767](https://jira.mongodb.org/browse/SWIFT-767) about that)

Reply:
went back to `close` over `shutdown` for the client and connection pool.

I did stick with `shutdown` for the executor because in that case it directly translates to calling 1 or possibly 2 `shutdown` methods on the thread pool and possibly ELG.